### PR TITLE
Laura & Jocelyn -- Carets

### PR DESCRIPTION
**Pull request by jocegonz**
# Scrabble
Congratulations! You're submitting your assignment.

## Comprehension Questions

|  Feature 	|   Feedback	|
|---	|---	|
|   Why do you think we chose to make the `score` method in the `Scoring` class a **class method** instead of an **instance method**? |  For testing purposes. We needed to be able to verify that the words entered were valid.  |
|   Describe an example of a test you wrote for a _nominal case_. |  We tested that the user gets a score based on a valid word enter. 	|
|   Describe an example of a test you wrote for an _edge case_. | To test if they won, we tested if 1 tested false at 99 points. It didn't, but when we added one more point, it returned true.  	|
|   What was your final test coverage percentage?	|   100%	|
|   Is there a specific piece of code you'd like feedback on?	|   We think we could have added more enumerables more to condense our code.	|

### CODE REVIEW

File: lib/scrabble.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            value += 5
+          when "j", "x"
+            value += 8
+          when "q", "z"
+            value += 10
+          end
+        end
+        return value
+      else
+        raise ArgumentError.new("Word has numbers")
```
Comment (CheezItMan) [inline lib/scrabble.rb:31]: Not necessarily just numbers.  Your regex matches any non-letter.

File: lib/scrabble.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          tie_breaker.push(key)
+        end
+      end
+
+      winner = ""
+      tie_breaker.each do |word|
+        if word.length == 7
+          winner = word
+        else
+          winner = tie_breaker.min{|a,b| a.size <=> b.size}
```
Comment (CheezItMan) [inline lib/scrabble.rb:55]: Good use of min here.

File: lib/player.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      total_score
+      return Scrabble::Scoring.score(word) # new instance of score
+    end
+
+    def total_score
+      sum_of_scores = []
+      @plays.each do |word|
+        sum_of_scores << Scrabble::Scoring.score(word)
+      end
+      @total_score = sum_of_scores.inject(:+)
```
Comment (CheezItMan) [inline lib/player.rb:30]: You can probably do this with:  

`      @total_score = @plays.map{ |word| Scrabble::Scoring.score(word)}.inject(:+)`

But yours works well.

File: lib/player.rb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      @tiles << tile_bag.draw_tiles(@plays.last.length)
+      @tiles = @tiles.flatten
+    end
+
+    private
+
+    def won?
+      if @total_score > 99
+        @won = true
+        return "You win!"
```
Comment (CheezItMan) [inline lib/player.rb:62]: This method makes more sense returning true or false, rather than a string.

File: specs/tilebag-spec.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
+require_relative "spec_helper"
```
Comment (CheezItMan) [inline specs/tilebag-spec.rb:1]: I would state that the filenames should be `snake_case` instead of with dashes.

**Comment by CheezItMan** [comment]
# Scrabble
## What We're Looking For

|  Feature  |   Feedback |
|---	|---	|
|  **General**	|   	|
|   Answered comprehension questions  |  Check, We made the `score` method as a class method because it doesn't have to do with a particular instance, so it makes sense as a class method.    |
|   Both Teammates contributed to the codebase	|  Commits from both team members.  |
|   Regular Commits with meaningful commit messages  | A lot of the commit both revolve around the waves completed.  I think you should instead focus on the functionality completed.  I would also do more commits.     |
|  **Scoring class**	|   	|
|  `score` method	|  Check  |
|   Uses appropriate data structure to store the letter score	|  You could do this much more elegantly with a hash, using the letter as the key and the score as the value.  |
|   Appropriately handles edge cases	|  Check, good use of ArgumentError and Regex.  |
|   Tests `score` edge cases (one-letter word, 7 letter word)	| Check   |
|  `highest_scoring_word` method	| Check   |
|   Appropriately handles edge cases	|  Check, handle 7-letter words, and words that tie.   |
|   Tests `highest_scoring_word` edge cases (ties, no words, no array)	|  I would also do a test varying the order of the words, and verifying that if two words have the same score and same length, the earlier word wins  |
|  **Player class**	|   	|
|  Uses an array to keep track of words played	|  Check  |
|  Uses existing code to create `highest_scoring_word` and `highest_word_score`	|  Check  |
|  Returns `true` or `false` from `won?` method | Check  |
|  Tests edge cases on the `play(word)` method for words that cannot be scored |   I don't see a test for when you play so many words that it brings the score over 100 (in that case `play` should return `false`).    |
|  **TileBag class**	|   	|
|  Uses a data structure to keep the set of default tiles in the bag	|  Check, good use of a hash, but you do have to reconstruct the array each time someone draws.    |
|  `draw_tiles` method uses the parameter to draw a set of random tiles |  Check  |
| `tiles_remaining` method returns a count	| Check, good use of inject.   |
|  Tests `draw_tiles` edge cases for different parameter inputs	|  Tests nominal cases, but doesn't test what happens when you draw all, or more than all, the tiles.    |
|  Summary   | Nice work overall, but you need to work more on throughly testing all the edge cases.    |

**Review by CheezItMan — COMMENTED** [review]
Not bad, just a few notes.
